packages/plugin-ext/src/main/browser/menus/menus-contribution-handler.ts

```diff
@@ -82,6 +107,81 @@ export class MenusContributionPointHandler {
         }
     }
 
+    protected registerScmTitleAction(menuAction: Menu): void {
+        const id = this.createSyntheticCommandId(menuAction, { prefix: '__plugin.scm.title.action.' });
+        const command: Command = { id };
+        this.commands.registerCommand(command, {
+            execute: () => this.commands.executeCommand(menuAction.command)
+        });
+
+        this.scmTitleCommandRegistry.registerCommand({ command: id, when: menuAction.when });
+
+        if (menuAction.group && menuAction.group !== 'navigation') {
+            const action: MenuAction = { commandId: id };
+            this.menuRegistry.registerMenuAction([...PATH, menuAction.group], action);
+        }
+
+        this.onDidRegisterCommand(menuAction.command, pluginCommand => {
+            command.iconClass = pluginCommand.iconClass;
+            command.category = menuAction.group;
+            command.label = pluginCommand.label;
+        });
+    }
+
+    protected registerScmResourceGroupAction(action: Menu): void {
+        const id = this.createSyntheticCommandId(action, { prefix: '__plugin.scm.resource.group.action.' });
+        const command: Command = { id };
+        this.commands.registerCommand(command, {
+            execute: arg => this.commands.executeCommand(action.command, arg)
+        });
+
+        const condition = action.when;
+        if (condition) {
+            let group = condition.substring(condition.indexOf('scmResourceGroup == ') + 20);
```

Comment:
It looks hacky. Cannot we add `ContextKey` for `scmResourceGroup` and use `when` condition as it is?

Reply:
We can not reach `scmResource` item here to set the key